Libraries/ConfigParser/src/config_parser.cpp

```diff
@@ -0,0 +1,178 @@
+// Windows headers
+#define WIN32_LEAN_AND_MEAN             // Exclude rarely-used stuff from Windows headers
+#include <windows.h>
+
+#include <fstream>
+
+#include "config_parser.h"
+#include "third_party/jsoncpp/source/include/json/json.h"
+
+using namespace StreamingToolkit;
+
+std::string ConfigParser::GetAbsolutePath(const std::string& file_name)
+{
+	TCHAR buffer[MAX_PATH];
+	GetModuleFileName(NULL, buffer, MAX_PATH);
+	std::string::size_type pos = std::string(buffer).find_last_of("\\/");
+	return std::string(buffer).substr(0, pos + 1) + file_name;
+}
+
+void ConfigParser::Parse(
+	const std::string& file_name,
+	ServerConfig* server_config,
+	ServiceConfig* service_config)
+{
+	memset(server_config, sizeof(ServerConfig), 0);
```

Comment:
this feels very old school, i'm not a big fan of managing our own memory (i know, it's just zeroing, but still) - what if we instead created the `ServerConfig` instance, and returned it?

from the caller perspective this cleans things up a bit, right?
```
auto configInstance = ConfigParser::Parse();
// use configInstance
```
over
```
ConfigType configInstance;
ConfigParser::Parse(&configInstance);
// use configInstance
```

Reply:
Not sure I understand your suggestion correctly but do you mean that we allocate ServerConfig object on heap and return it?

Reply:
yeah that's what i'm proposing, curious to hear your thoughts

Reply:
I wouldn't recommend that since it's potential for memory leak. I usually forget to deallocate object returned by a method. In this case, I'm afraid that devs might forget to call "delete configInstance". Does that make sense to you?

Reply:
i guess i'd use a `std::shared_ptr` if i did it the way i'm proposing, but yeah i see your point that could be very easy to forget.